File: c++/2270.cpp
// Link: https://leetcode.com/problems/number-of-ways-to-split-array/description/
// Runtime: 0 ms
// Memory: 89.32 MB

class Solution {
public:
  int waysToSplitArray(const std::vector<int> &nums) {
    long long leftSum = 0, rightSum = 0;
    for (const auto &val : nums)
      rightSum += val;

    int numberOfWaysToSplitArray = 0;
    for (int i = 0; i < static_cast<int>(nums.size() - 1); i++) {
      leftSum += nums[i];
      rightSum -= nums[i];

      if (leftSum >= rightSum)
        numberOfWaysToSplitArray++;
    }

    return numberOfWaysToSplitArray;
  }
};

File: python/2270.py
# Link: https://leetcode.com/problems/number-of-ways-to-split-array/description/
# Runtime: 48 ms
# Memory: 32.34 MB


class Solution:
    def waysToSplitArray(self, nums: List[int]) -> int:
        left_sum, right_sum = 0, sum(nums)

        number_of_ways_to_split_array = 0
        for i in range(len(nums) - 1):
            left_sum += nums[i]
            right_sum -= nums[i]

            if left_sum >= right_sum:
                number_of_ways_to_split_array += 1

        return number_of_ways_to_split_array
